src/api/v2/compatibility/CompatibleAzExtTreeDataProvider.ts

```diff
@@ -0,0 +1,108 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, compatibilitySubscriptionExperience, contextValueExperience, findByIdExperience, IActionContext, IFindTreeItemContext, isWrapper, ITreeItemPickerContext } from "@microsoft/vscode-azext-utils";
+import { Disposable, Event, TreeItem, TreeView } from "vscode";
+import { SubscriptionTreeItem } from "../../../tree/SubscriptionTreeItem";
+import { ResourceGroupsItem } from "../../../tree/v2/ResourceGroupsItem";
+import { ResourceTreeDataProviderBase } from "../../../tree/v2/ResourceTreeDataProviderBase";
+
+/**
+ * An intermediate class that exists just to redeclare several events as abstract, so they
+ * can be re-redeclared as a accessors in {@link CompatibleAzExtTreeDataProvider} below
+ */
+abstract class IntermediateCompatibleAzExtTreeDataProvider extends AzExtTreeDataProvider {
+    public abstract onDidChangeTreeData: Event<AzExtTreeItem | undefined>;
+    public abstract onTreeItemCreate: Event<AzExtTreeItem>;
+    public abstract onDidExpandOrRefreshExpandedTreeItem: Event<AzExtTreeItem>;
+}
+
+export class CompatibleAzExtTreeDataProvider extends IntermediateCompatibleAzExtTreeDataProvider {
+    public constructor(private readonly tdp: ResourceTreeDataProviderBase) {
+        super({} as unknown as AzExtParentTreeItem, undefined as unknown as string);
+    }
+
+    //#region Things that should not be called
+    public override trackTreeItemCollapsibleState(_treeView: TreeView<AzExtTreeItem>): Disposable {
+        throw new Error('This method should never be called');

```

Comment:
I know that these are supposed to never be called, but would it be worth outputting what method or accessor was called to throw the error, if an internal error ever does happen?

Reply:
I think that's a good idea. Call stacks frequently end up being useless thanks to webpack.